Gems/ROS2/Code/Source/Manipulation/JointsManipulationEditorComponent.h

```diff
@@ -32,10 +32,12 @@ namespace ROS2
         // AzToolsFramework::Components::EditorComponentBase overrides
         void BuildGameEntity(AZ::Entity* gameEntity) override;
 
+        void Activate();
+
     private:
         PublisherConfiguration m_jointStatePublisherConfiguration;
         AZStd::unordered_map<AZStd::string, float> m_initialPositions;
```

Comment:
How about replacing of these two collections with 
AZStd::vector<AZStd::pair<<AZStd::string, float>> and letting user reorder items in UI?
or AZStd::map from index to such pair if it is easier?

Reply:
Unfortunately, I had problems with reordering elements of vectors: https://github.com/o3de/o3de/issues/16648.

Reply:
There is a `set<{name, position, index}>`. It keeps all the information in one place and has nice input method: 
![image](https://github.com/o3de/o3de-extras/assets/138626322/01b372e9-9dd4-414b-88ab-32676b9a6086)


But because of that I see no way of straightforward migration of prefab overrides (as it would require knowledge of the prefab).